backend/python/app/services/interfaces/goal_service.py

```diff
@@ -45,4 +45,14 @@ def add_new_goal(self, goal, type):
         :rtype: GoalDTO or None
         :raises Exception: if goal type is invalid
         """
+
+    @abstractmethod
+    def get_goals_by_intake(self, intake_id, type=None):
+        """Get the goals associated with a given intake id;
+        If goal type specified, only return goals of given type
+        :param intake_id: int of intake ID
+        :return: List of goals for given intake
+        :rtype: List of GoalDTOs
+        :raises Exception: if goal or intake is invalid
```

Comment:
nit: maybe something like 
```suggestion
        :raises Exception: If intake ID does not exist
```